yarn-project/circuits.js/src/structs/avm/avm.ts

```diff
@@ -280,6 +270,99 @@ export class AvmContractInstanceHint {
   }
 }
 
+export class AvmContractBytecodeHints {
+  /*
+   * @param bytecode currently the bytecode of the nested call function, will be changed to the contract bytecode (via the dispatch function) of the nested call
+   * @param contractInstance the contract instance of the nested call, used to derive the contract address
+   * @param contractClassPreimage the contract class preimage of the nested call, used to derive the class id
+   * */
+  constructor(
+    public readonly bytecode: Buffer,
+    public contractInstanceHint: AvmContractInstanceHint,
+    public contractClassHint: ContractClassIdPreimage,
```

Comment:
We are definitely "over-hinting" since we also transmit the address, bytecode commitment and class id.

But for now these are useful in checking that the values calculated in witgen match the simulator